app/models/landing_page.rb

```diff
@@ -4,13 +4,16 @@ class LandingPage < ContentItem
   ADDITIONAL_CONTENT_PATH = "lib/data/landing_page_content_items".freeze
 
   def initialize(content_store_response)
-    if content_store_response.dig("details", "blocks")
-      super(content_store_response)
-    else
-      super(content_store_response.deep_merge(load_additional_content(content_store_response["base_path"])))
+    override_content_store_hash = nil
+    if content_store_response.dig("details", "blocks").nil?
+      override_content_store_hash = content_store_response.to_hash.deep_merge(
```

Comment:
In the tests a hash converted to a `GdsApi::Response`. Is it possible for the same to be done here?
i.e. after the `content_store_response` is converted to a hash to allow the `deep_merge` for the additional content, could it then be converted back to a `GdsApi::Response` before being passed to the super class?

I'm trying to think of how to limit the blast radius for something that will only temporarily affect Landing Pages and that will be easier to unpick later.

Reply:
The `GdsApi::Response` constructor takes a `RestClient::Response` as its input. The tests construct a fake one, but that's okay because we can use an `instance_double`. To do it here, we'd need to make a real fake HTTP response by turning the hash back to a JSON string for the body and reassembling the HTTP headers from the original response.

Super gross, basically.

I don't think the blast radius of this change is particularly bad? We're adding an optional parameter to the constructor, and it's only provided in this one case. Other than that, the logic should be unchanged.

Reply:
I did consider a couple of other options, which you may find more or less gross than this:

Reply:
1) Wrap the `GdsApi::Response` with another class which intercepts any method calls which return things from the body and merges in the additional content. Pass this into the ContentItem constructor, and it doesn't need to change.

Reply:
2. Monkey patch the `GdsApi::Response` instance 🙈

Reply:
I felt this was the least magical / most easy to reason about option.

Reply:
Yes, this is probably the least ugly.

I think it would be a bit nicer if the truthy check `if content_store_response.dig("details", "blocks")` was kept in and it was only the `else` block that had the overhaul. 

i.e.
```ruby
def initialize(content_store_response)  
  if content_store_response.dig("details", "blocks")
    super(content_store_response)
  else 
    override_content_store_hash = content_store_response.to_hash
    override_content_store_hash.deep_merge(
      load_additional_content(content_store_response["base_path"]),
    )

    super(content_store_response, override_content_store_hash:)
  end

  @blocks = (content_store_hash.dig("details", "blocks") || []).map { |block_hash| BlockFactory.build(block_hash) }
end
```

But again not a blocker.